### ISSUE TITLE
`masked_select` on Z-Normalization slower than indexing

### ISSUE THREAD

**Issue by ramonemiliani93** [fepegar/torchio#1017]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug summary

The torch function `masked_select` is slower than indexing:

```
In [1]: import torch

In [2]: x = torch.rand(300, 300, 300)

In [3]: y = torch.rand(300, 300, 300) > 0.5

In [4]: %timeit -n 20 x[y]
102 ms ± 2.33 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)

In [5]: %timeit -n 20 x.masked_select(y)
152 ms ± 2.17 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)
```

### Code for reproduction

```python
In [1]: import torch

In [2]: x = torch.rand(300, 300, 300)

In [3]: y = torch.rand(300, 300, 300) > 0.5

In [4]: %timeit -n 20 x[y]
102 ms ± 2.33 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)

In [5]: %timeit -n 20 x.masked_select(y)
152 ms ± 2.17 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)
```

### Actual outcome

Faster indexing with `x[y]`

### Expected outcome

Faster normalization.

### System info

```shell
Platform:   macOS-13.1-x86_64-i386-64bit
TorchIO:    0.18.86
PyTorch:    1.13.1
SimpleITK:  2.2.1 (ITK 5.3)
NumPy:      1.24.1
Python:     3.9.13 (main, Jul 27 2022, 15:28:09)
[Clang 13.0.0 (clang-1300.0.29.30)]
```

**Comment by fepegar** [comment]
https://github.com/all-contributors please add @ramonemiliani93 for code

**Comment by fepegar** [comment]
@all-contributors please add @ramonemiliani93 for code
